### PR TITLE
round IndexAccess for numerical strings over int columns

### PR DESCRIPTION
**Pull request by jycor**
fixes https://github.com/dolthub/dolt/issues/7261

### CODE REVIEW

File: enginetest/queries/queries.go [hunk 1]
```diff
@@ -8719,6 +8719,14 @@ from typestable`,
 			{1, 0, 3, 2},
 		},
 	},
+	{
+		Query: "select * from mytable where i > '-0.5';",
```
Comment (max-hoffman) [inline enginetest/queries/queries.go:8723]: I tried out index lookup but I'm not seeing the same issue for some reason, does that make sense that it's just static range scans?
```
select /*+ LOOKUP_JOIN(b,a) */ * from mytable a join mytable b on a.i = '-.5' and a.s = b.s
Project
 ├─ columns: [a.i:2!null, a.s:3!null, b.i:0!null, b.s:1!null]
 └─ LookupJoin
     ├─ Eq
     │   ├─ a.s:3!null
     │   └─ b.s:1!null
     ├─ TableAlias(b)
     │   └─ Table
     │       ├─ name: mytable
     │       ├─ columns: [i s]
     │       ├─ colSet: (3,4)
     │       └─ tableId: 2
     └─ Filter
         ├─ Eq
         │   ├─ a.i:0!null
         │   └─ -.5 (longtext)
         └─ TableAlias(a)
             └─ IndexedTableAccess(mytable)
                 ├─ index: [mytable.i]
                 ├─ keys: [-.5 (longtext)]
                 ├─ colSet: (1,2)
                 ├─ tableId: 1
                 └─ Table
                     ├─ name: mytable
                     └─ columns: [i s]
```

**Review by max-hoffman — APPROVED** [review]
LGTM, seems like there is potential for adjacent tests to make sure index conversions work for different types

File: enginetest/queries/queries.go (hunk 1 above)
Comment (jycor) [reply enginetest/queries/queries.go:8723]: I believe it hits this case if we are using a lookup join:
https://github.com/dolthub/go-mysql-server/pull/2246/files#diff-66b16c24d40520a9c552a4093466441a1623797f6aea41f208809bf2eee3de8cR114

(The range is empty unless its a whole number)

Comment (max-hoffman) [reply enginetest/queries/queries.go:8723]: kk, so I guess this bug only applies to non-equalities.

**Pull request merged**